src/smartsplit/components/admin-list.js

```diff
@@ -13,33 +10,38 @@ import { Colors } from "../../theme"
 import { TouchableWithoutFeedback, View } from "react-native"
 import { Platform } from "../../platform"
 import Hourglass from "../../svg/hourglass"
-import { AdminListMenu, DefaultMenu, PendingMenu } from "./admin-list-menus"
+import {
+	AdminListMenu,
+	DefaultMenu,
+	PendingMenu,
+	SimpleMenu,
+} from "./admin-list-menus"
 
 export const AdminListStyle = StyleSheet.create({
 	frame_pending: {
 		backgroundColor: Colors.background.underground,
 	},
 })
 
-function getTitle(children) {
-	let title
-	if (typeof children === "object") {
-		title = []
+function formatContent(children) {
```

Comment:
Elle sert à quoi cette fonction? Simplement ajouter le `key` ?

Quel est le problème original?

Reply:
L'idée c'était de pouvoir mettre n'importe quel type de contenu dans une row de la liste. Ca peut être une simple string, un array d'objets ou carrément un fragment React. Dans les deux derniers cas je trouve qu'assigner automatiquement des keys au moment du render c'est plus neat que de devoir renseigner lesdites key pour chaque objet au niveau du parent. Selon moi c'est pas de la responsabilité du component parent de gérer les clés des items d'une liste enfant

Reply:
Hmm, il y a quelques problèmes avec ça. Là tu te base sur l'index de l'array, ce qui [est un antipattern pour les keys dans React](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). L'idée des key c'est de pouvoir dire à React quel élément correspond à quoi, même si l'ordre des éléments change. Avec l'index de l'array, c'est équivalent à ne pas mettre les keys. C'est à la composante parente de spécifier ses key pour les éléments passés à la composante, car ta liste elle n'a aucune idée sur quoi les key devraient être basés. Pour nos listes par exemple, ça devrait être le UUID de l'item.

Mais sinon, si jamais tu veux éviter les keys, il y a la fonction `mapFragmentChildren` qui va wrap tout ça dans un fragment et enlever les warnings complètement. Je pense que ça serait à la composante parente d'utiliser cette fonction avant de le donner à la liste.

C'est bien d'être neat, mais faut faire attention pour pas être _trop_ neat :)